File: src/main/java/cn/cdtft/ideaplugin/androideventbus/SenderFilter.java
package cn.cdtft.ideaplugin.androideventbus;

import com.intellij.psi.*;
import com.intellij.usages.Usage;
import com.intellij.usages.UsageInfo2UsageAdapter;

/**
 * SenderFilter
 *
 * @author john
 * @since 2019-04-13
 */
public class SenderFilter implements Filter {
    private final PsiElement mTagPsiElement;

    public SenderFilter(PsiElement tagPsiElement) {
        mTagPsiElement = tagPsiElement;
    }

    @Override
    public boolean shouldShow(Usage usage) {
        PsiElement element = ((UsageInfo2UsageAdapter) usage).getElement();
        if (element instanceof PsiReferenceExpression) {
            if ((element = element.getParent()) instanceof PsiMethodCallExpression) {
                PsiMethodCallExpression callExpression = (PsiMethodCallExpression) element;
                final PsiExpression[] expressions = callExpression.getArgumentList().getExpressions();
                for (PsiExpression exp : expressions) {
                    if (exp.getLastChild().getText().equals(mTagPsiElement.getLastChild().getText())) {
                        return true;
                    }
                }
            }
        }

        return false;
    }
}

File: src/main/kotlin/cn/cdtft/plugin/aep/SenderFilter.kt
package cn.cdtft.plugin.aep

import com.intellij.psi.PsiElement
import com.intellij.psi.PsiExpression
import com.intellij.psi.PsiMethodCallExpression
import com.intellij.psi.PsiReferenceExpression
import com.intellij.usages.Usage
import com.intellij.usages.UsageInfo2UsageAdapter

/**
 * SenderFilter
 *
 * @author john
 * @since 2019-04-13
 */
class SenderFilter internal constructor(private val mTagPsiElement: PsiElement) : Filter {
    override fun shouldShow(usage: Usage): Boolean {
        var element = (usage as UsageInfo2UsageAdapter).element
        if (element is PsiReferenceExpression) {
            if ((element.parent.also { element = it }) is PsiMethodCallExpression) {
                val callExpression: PsiMethodCallExpression = element as PsiMethodCallExpression
                val expressions: Array<PsiExpression> = callExpression.argumentList.expressions
                for (exp in expressions) {
                    if (exp.lastChild.text.equals(mTagPsiElement.lastChild.text)) {
                        return true
                    }
                }
            }
        }

        return false
    }
}
